code-challenges.test.js

```diff
@@ -15,47 +19,323 @@
+// // Expected output: false
+
+
+//Pseudo Code:
+/*
+-Input an array, function checks for a unique element that repeats twice and another unique element that repeats three times, outputs a boolean true or false.
+-Perform a .reducer to check and track number of times a value is repeated
+-Use Object.values to convert the object from reduce to an array
+-Check if the output array contains a 2 && 3. if statements
+-In addition, need to do a red-green-refactor methodolgy, hand1, hand2, and hand3.
+*/
+
+// // First Step: Perform Red
+
+// describe("fullHouse", () => {
+    
+//     it("takes in an array of 5 numbers and determines whether or not the array is a “full house", () => {
+
+//         expect(fullHouse(hand1)).toEqual(true)
+//     })
+// })
+
+// Received an error =>  ReferenceError: fullHouse is not defined
+
+//Second Step: Make the test pass for hand1, hand2, hand3.
+// describe("fullHouse", () => {
+
+//     const hand1 = [5, 5, 5, 3, 3]
+//     // Expected output: true
+//     const hand2 = [5, 5, 3, 3, 4]
+//     // Expected output: false
+//     const hand3 = [5, 5, 5, 5, 4]
+//     // Expected output: false
+
+//     it("takes in an array of 5 numbers and determines whether or not the array is a “full house", () => {
+
+//         expect(fullHouse(hand1)).toEqual(true)
+//         expect(fullHouse(hand2)).toEqual(false);
+//         expect(fullHouse(hand3)).toEqual(false);
+//     })
+// })
+
+
 
 
 
 // b) Create the function that makes the test pass.
+// -Input an array, function checks for a unique element that repeats twice and another unique element that repeats three times, outputs a boolean true or false.
+// -Perform a .reducer to check and track number of times a value is repeated
+// -Use Object.values to convert the object values from reduce to an array
+// -Check if the output array contains a 2 && 3. if statements
+
+// const fullHouse = (hand) => {
+
+//     let tracker = hand.reduce((uniqueValues,currentValue)=>{
+        
+//         if(uniqueValues[currentValue]){
+//             uniqueValues[currentValue] = uniqueValues[currentValue] + 1
+//         }
+//         else{
+//             uniqueValues[currentValue] = 1
+//         }
+//         return uniqueValues
+//     },{})
+
+//    
+//     if(Object.values(tracker).includes(2) && Object.values(tracker).includes(3)){
+//         return true
+//     }
+//     else{
+//         return false
+//     }
+// }
+
+
+
+//Third Step: Refactor
+/*
+Reformat by having terinary operators and added a Object.hasOwn to check if property exists.
+*/
+
+const fullHouse = (hand) => {
+
+    let tracker = hand.reduce((uniqueValues, currentValue) => {
+        Object.hasOwn(uniqueValues, currentValue) ? uniqueValues[currentValue] = uniqueValues[currentValue] + 1 : uniqueValues[currentValue] = 1;
+        return uniqueValues
+    },{})
+
+    return (Object.values(tracker).includes(2) && Object.values(tracker).includes(3)) ? true : false
+    
+}
+
```

Comment:
Quite the interesting approach! Just a tiny thought - you might be able to refactor `uniqueValues[currentValue] = uniqueValues[currentValue] + 1` to `uniqueValues[currentValue] += 1`